File: solved/UVa855/855.py
#!/usr/bin/env python3
import sys

try:
    T = int (sys.stdin.readline ())

    for t in range (T):
        l = sys.stdin.readline ()
        l = l.rstrip ('\n')

        [S, A, F] = [ int (x) for x in l.split () ]
        coord = [[], []]
        try:
            for f in range (F):
                l = sys.stdin.readline ()
                l = l.rstrip ('\n')
                [s, a] = [ int (x) for x in l.split () ]
                coord[0].append (s)
                coord[1].append (a)

            s = [0, 0]
            for i in [0, 1]:
                coord[i].sort ()
                s[i] = coord[i][(F - 1) // 2]

            print ("(Street: " + str (s[0]) + ", Avenue: " + str (s[1]) + ")")
        except:
            pass
except:
    pass

File: solved/UVa855/855.cpp
#include <bits/stdc++.h>
using namespace std;

int
main ()
{
	int T;
	for (scanf ("%d", &T); T--;) {
		long S, A, F;
		vector<int> v[2];

		scanf ("%ld%ld%ld", &S, &A, &F);

		for (int f = 0; f < F; f++) {
			long s, a;
			scanf ("%ld%ld", &s, &a);
			v[0].push_back (s);
			v[1].push_back (a);
		}
		
		long s, a;

		sort (v[0].begin (), v[0].end ());
		sort (v[1].begin (), v[1].end ());

		s = v[0][(F - 1)/2];
		a = v[1][(F - 1)/2];

		printf ("(Street: %ld, Avenue: %ld)\n", s, a);
	}

	return 0;
}
